ush/setup.py

```diff
@@ -1490,72 +1490,53 @@ def dict_find(user_dict, substring):
-    # SDF_USES_RUC_LSM to True or False accordingly.
-    #
-    # -----------------------------------------------------------------------
-    #
-    workflow_config["SDF_USES_RUC_LSM"] = check_ruc_lsm(
-        ccpp_phys_suite_fp=ccpp_phys_suite_in_ccpp_fp
-    )
-    #
-    # -----------------------------------------------------------------------
-    #
-    # Check if the Thompson microphysics parameterization is being
-    # called by the physics suite and modify certain workflow arrays to
-    # ensure that fixed files needed by this parameterization are copied
-    # to the FIXam directory and appropriate symlinks to them are
-    # created in the run directories. Set the boolean flag
-    # SDF_USES_THOMPSON_MP to indicates whether Thompson MP is called by
-    # the physics suite.
+    # Read CCPP suite definition file and perform actions based on its
+    # contents as necessary
     #
     # -----------------------------------------------------------------------
     #
 
-    link_thompson_climo = (
-        get_extrn_ics["EXTRN_MDL_NAME_ICS"] not in ["HRRR", "RAP"]
-    ) or (get_extrn_lbcs["EXTRN_MDL_NAME_LBCS"] not in ["HRRR", "RAP"])
-    use_thompson, mapping, fix_files = set_thompson_mp_fix_files(
-        ccpp_phys_suite_fp=ccpp_phys_suite_in_ccpp_fp,
-        thompson_mp_climo_fn=workflow_config["THOMPSON_MP_CLIMO_FN"],
-        link_thompson_climo=link_thompson_climo,
-    )
+    ccpp_suite_xml = load_xml_file(workflow_config["CCPP_PHYS_SUITE_IN_CCPP_FP"])
 
-    workflow_config["SDF_USES_THOMPSON_MP"] = use_thompson
+    # Need to track if we are using RUC LSM for the make_ics step
+    workflow_config["SDF_USES_RUC_LSM"] = has_tag_with_value(ccpp_suite_xml, "scheme", "lsm_ruc")
+
+    # Thompson microphysics needs additional input files and namelist settings
+    workflow_config["SDF_USES_THOMPSON_MP"] = has_tag_with_value(ccpp_suite_xml, "scheme", "mp_thompson")
+
+    if workflow_config["SDF_USES_THOMPSON_MP"]:
+    
+        logging.debug(f'Selected CCPP suite ({workflow_config["CCPP_PHYS_SUITE"]}) uses Thompson MP')
+        logging.debug(f'Linking additional fix files')
+        # Add thompson-specific fix files to CYCLEDIR_LINKS_TO_FIXam_FILES_MAPPING and
+        # FIXgsm_FILES_TO_COPY_TO_FIXam; see config_defaults.yaml for more info on these variables
+        thompson_fix_files = [
+            "CCN_ACTIVATE.BIN",
+            "freezeH2O.dat",
+            "qr_acr_qg.dat",
+            "qr_acr_qs.dat",
+            "qr_acr_qgV2.dat",
+            "qr_acr_qsV2.dat",
+        ]
+        # If the model ICs or BCs are not from RAP or HRRR, they will not contain aerosol
+        # climatology data needed by the Thompson scheme, so we need to provide a separate file
+        if (get_extrn_ics["EXTRN_MDL_NAME_ICS"] not in ["HRRR", "RAP"] or
+           get_extrn_lbcs["EXTRN_MDL_NAME_LBCS"] not in ["HRRR", "RAP"]):
+            thompson_fix_files.append(workflow_config["THOMPSON_MP_CLIMO_FN"])
```

Comment:
Why does this one (`THOMPSON_MP_CLIMO_FN `) get parameterized, but all others are hard-coded here in the code?

Reply:
This file is used in a different context than the rest of the fix files, so it does need to be separate. But you bring up a good point, I have moved this array definition to `parm/fixed_files_mapping.yaml` with the rest of the fix file lists.